### DISCUSSION TITLE
On deleting a task I get - Hook Error: Expected feedback from a failing hook script.

### DISCUSSION THREAD

**Discussion by jedmallen** [GothenburgBitFactory/taskwarrior#2697] [Community support] [answered]
I am not sure if this is a bug as the same error in Issues is from 2015.

Background, I recently converted from system python2 to using pyenv and python -V is 3.9.9 and python2 is for 2.7.18.

Environment: WSL2 Windows 10

**Comment by tbabej** [comment, marked as answer]
This error comes out if a hook fails. My best guess is that your have a Python-based hook present, and it uses a library that was installed under python2, but is not installed under python3.

You can see your currently active hooks by running `task diag`. I'd suggest double checking that all dependencies are present under python3.

**Reply by jedmallen** [reply]
thanks @tbabej 

seems it's my timewarrior hook

Active: on-modify.timewarrior (executable)

**Reply by jedmallen** [reply]
hi @tbabej ,

I examined the on-modify.timewarrior script and it is python3 and it's not corrupted in any way.

I did a `task rc.debug.hooks=2 20 start` and I got the following which I does not point to anything wrong.

```
task rc.debug.hooks=2 20 start  
Timer Config::load (/home/user/.taskrc) 0.000582 sec
No context.
Found hook script /home/user/.task/hooks/on-modify.timewarrior
Filtered 37 tasks --> 1 tasks [pending only]
Task::modify
Hook: Calling /home/user/.task/hooks/on-modify.timewarrior
Hook: input
  {"description":"create hms urls repo","entry":"20211025T014159Z","modified":"20211025T014159Z","status":"pending","uuid":"f68b60b2-c19b-4096-8ac8-4fbe3c81c720"}
  {"description":"create hms urls repo","entry":"20211025T014159Z","modified":"20211025T014159Z","start":"20211202T095711Z","status":"pending","uuid":"f68b60b2-c19b-4096-8ac8-4fbe3c81c720"}
Hooks: args
  api:2
  args:task rc.debug.hooks=2 20 start
  command:start
  rc:/home/user/.taskrc
  data:/home/user/.task
  version:2.5.1
Timer Hooks::execute (/home/user/.task/hooks/on-modify.timewarrior) 0.000745 sec
Hook: output
Hook: Completed with status 255

Configuration override rc.debug.hooks:2
Hook Error: Expected feedback from a failing hook script.
```

**Reply by jedmallen** [reply]
This is resolved. When I installed pyenv, python3 is pointing to ~/.pyenv/shims/python3 but the on-modify script's shebang line points to /usr/bin/python3. Just needed to make a softlink to the pyenv python3 in /usr/bin/python3. :)
